### PR TITLE
Add python to OSX CONTRIBUTING requirements

### LINKED ISSUE #4080: Regression tests are broken on OS X
On OS X, I followed the instructions to contribute to Citus: https://github.com/citusdata/citus/blob/master/CONTRIBUTING.md

The last step to run the tests fails with the following message.

<pre>
ozgun$ make check
...
============== installing citus                       ==============
CREATE EXTENSION
============== running regression test queries        ==============
test multi_extension              ... env: python3: No such file or directory
diff command failed with status 32512: diff  "/workplace/citus/src/test/regress/expected/multi_extension.out" "/workplace/citus/src/test/regress/results/multi_extension.out" > "/workplace/citus/src/test/regress/results/multi_extension.out.diff"
waiting for server to shut down.... done
</pre>

I went back a few releases to see when we broke the regression tests.  If I build from our citus-9.3 branch, I see the above failure. If I build from citus-9.1, all regression tests pass.

### PR DESCRIPTION
**Pull request by JelteF**
Fixes #4080

### CODE REVIEW

**Comment by metdos** [comment]
Shall we also add --with-icu to contributing? https://github.com/citusdata/citus/issues/3278

**Comment by JelteF** [comment]
> Shall we also add --with-icu to contributing? #3278

This only applies if you build Postgres from source, which the instructions in CONTRIBUTING don't do.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ why we ask this as well as instructions for how to proceed, see the
 
   ```bash
   brew update
-  brew install git postgresql
+  brew install git postgresql python
```
[the remark this answers is not in the document]
Comment (JelteF) [reply CONTRIBUTING.md:23]: I don't really think such details are necessary. Also, most people will have python installed already.

[the remark this answers is not in the document]
Comment (SaitTalhaNisanci) [reply CONTRIBUTING.md:23]: I am guessing that the only reason we need python is for failure tests specifically for MITM. But for those tests to pass we need to create a virtual env and install dependencies, without doing that `make check-failure` will fail. So if someone follows the contribution README, they will still see failures, I would expect to not get any failures when following the README in tests, what do you think?

Also I am not sure if we need python for any other place.

[the remark this answers is not in the document]
Comment (JelteF) [reply CONTRIBUTING.md:23]: The `diff-filter` tool we have uses `python3`, so it's required for all tests.

**Review by SaitTalhaNisanci — APPROVED** [review]
I think it might make sense to put a link to how to run failure tests so that users know why `make check` fails. Because the issue this PR fixes is about `regression tests not passing`, and they will still not pass with a simple `make check`. At least a link like this: https://github.com/citusdata/citus/blob/master/src/test/regress/mitmscripts/README.md

**Pull request merged**